### PR TITLE
Overall minors

### PR DESCRIPTION
**Pull request by liketurbo**
- Unignore server errors
- ErrorMessage component
- Get rid of React key warning
- Set placeholder in DashboardHeader

### CODE REVIEW

File: api/src/kafka/kafka.ts [hunk 1]
```diff
@@ -1,18 +1,19 @@
-import { Kafka } from "kafkajs";
+import { Kafka, Partitioners } from "kafkajs";
 
 const kafka = new Kafka({
   clientId: "my-app",
   brokers: ["localhost:9092"],
 });
 
-export const kafkaProducer = kafka.producer();
+export const kafkaProducer = kafka.producer({ createPartitioner: Partitioners.LegacyPartitioner });
```
Comment (benawad) [inline api/src/kafka/kafka.ts:8]: What does this do?

File: web/src/app/ui/EditableTextField.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, { ChangeEvent, useEffect, useRef, useState } from "react";
 import { Input } from "./Input";
 
 interface EditableTextFieldProps {
-  text: string;
+  text?: string;
```
Comment (benawad) [inline web/src/app/ui/EditableTextField.tsx:5]: This component should always take some text even if its an empty string

File: api/src/kafka/kafka.ts (hunk 1 above)
Comment (liketurbo) [reply api/src/kafka/kafka.ts:8]: This line is actually related to the warning:

```json
{"level":"WARN","timestamp":"2024-02-03T17:02:58.261Z","logger":"kafkajs","message":"KafkaJS v2.0.0 switched default partitioner. To retain the same partitioning behavior as in previous versions, create the producer with the option \"createPartitioner: Partitioners.LegacyPartitioner\". See the migration guide at https://kafka.js.org/docs/migration-guide-v2.0.0#producer-new-default-partitioner for details. Silence this warning by setting the environment variable \"KAFKAJS_NO_PARTITIONER_WARNING=1\""}
```

So to answer your question:

> To retain the same partitioning behavior as in previous versions

Comment (liketurbo) [reply api/src/kafka/kafka.ts:8]: Probably it would better to set `DefaultPartitioner` v2, as it is a new project 🤔

Comment (benawad) [reply api/src/kafka/kafka.ts:8]: Should we not use JavaCompatiblePartitioner?

Comment (benawad) [reply api/src/kafka/kafka.ts:8]: Yeah

**Pull request merged**